### PR TITLE
Convert SimpleBrickFragment to kotlin

### LINKED ISSUE #208: SimpleBrickFragment should be in kotlin

### PR DESCRIPTION
**Pull request by thusson13**
Fixes https://github.com/wayfair/brickkit-android/issues/208

### CODE REVIEW

File: BrickKit/app/src/main/java/com/wayfair/brickkitdemo/mainfragment/MainActivityFragment.kt [hunk 1]
```diff
@@ -30,7 +30,7 @@ class MainActivityFragment : BrickFragment() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        val placeholderBrick = ActiveBrick(resources, R.string.empty) { changeFragment(SimpleBrickFragment()) }
+        val placeholderBrick = ActiveBrick(resources, R.string.empty) { changeFragment(SimpleBrickFragment.newInstance(100)) }
```
Comment (kenyee) [inline BrickKit/app/src/main/java/com/wayfair/brickkitdemo/mainfragment/MainActivityFragment.kt]: why 100? should it be a constant?

File: BrickKit/app/src/main/java/com/wayfair/brickkitdemo/simplebrick/SimpleBrickFragment.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    companion object {
+        /**
+         * Create a new instance of a SimpleBrickFragment.
+         *
+         * @param numberOfBricks The number of bricks you want in the fragment
+         * @return The SimpleBrickFragment you created
+         */
+        fun newInstance(numberOfBricks: Int = 100) = SimpleBrickFragment().apply {
+            this.numberOfBricks = numberOfBricks
```
Comment (kenyee) [inline BrickKit/app/src/main/java/com/wayfair/brickkitdemo/simplebrick/SimpleBrickFragment.kt]: should use frag args or it won't handle app resurrection

File: BrickKit/app/src/main/java/com/wayfair/brickkitdemo/mainfragment/MainActivityFragment.kt (hunk 1 above)
Comment (thusson13) [reply BrickKit/app/src/main/java/com/wayfair/brickkitdemo/mainfragment/MainActivityFragment.kt]: this was unnecessary so I removed it (but made the underlying value a constant)

File: BrickKit/app/src/main/java/com/wayfair/brickkitdemo/simplebrick/SimpleBrickFragment.kt (hunk 2 above)
Comment (thusson13) [reply BrickKit/app/src/main/java/com/wayfair/brickkitdemo/simplebrick/SimpleBrickFragment.kt]: Done

File: BrickKit/app/src/main/java/com/wayfair/brickkitdemo/simplebrick/datamodel/TextDataModel.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    var text: String = initialText
+        private set(value) {
+            field = value
+            notifyChange()
+        }
+
+    /**
+     * Append the string to the existing text.
+     *
+     * @param newText the new sting to be appended
```
Comment (kenyee) [inline BrickKit/app/src/main/java/com/wayfair/brickkitdemo/simplebrick/datamodel/TextDataModel.kt]: nit: typo should be "string"

**Review by kenyee — APPROVED** [review]

File: BrickKit/app/src/main/java/com/wayfair/brickkitdemo/simplebrick/datamodel/TextDataModel.kt (hunk 3 above)
Comment (thusson13) [reply BrickKit/app/src/main/java/com/wayfair/brickkitdemo/simplebrick/datamodel/TextDataModel.kt]: fixed

**Pull request merged**